### ISSUE TITLE
Support for remote Azurite

### ISSUE THREAD

**Issue by Hardtack** [Azure/azure-storage-python#478]
### Which service(blob, file, queue) does this issue concern?
All services

### Which version of the SDK was used? Please provide the output of `pip freeze`.
```
azure-common==1.1.11
azure-nspkg==2.0.0
azure-storage-blob==1.3.0
azure-storage-common==1.3.0
azure-storage-nspkg==3.0.0
```

### What problem was encountered?
I'm trying to connect remote [Azurite](https://github.com/Azure/Azurite), not in localhost, but _azure-storage-python_ does not allow to use `is_emulated=True` with `custom_domain`, it forces to use `custom_domain=127.0.0.1` when `is_emulated=True`.
What I actually want to do is connecting to Azurite that is running on Docker container run by _docker-compose_ but I can't.

### Have you found a mitigation/solution?
Please allow to use custom domain with `is_emulated=True`

**Comment by zezha-msft** [comment]
@Hardtack, Ok I see. I'll look into improving the SDK's user experience for Azurite. Thanks!

**Comment by zezha-msft** [comment]
Fixed in 1.4.0.
